File: javav2/example_code/mediastore/src/main/java/com/example/mediastore/CreateContainer.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.mediastore;

// snippet-start:[mediastore.java2.create_container.main]
// snippet-start:[mediastore.java2.create_container.import]
import software.amazon.awssdk.services.mediastore.MediaStoreClient;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.mediastore.model.CreateContainerRequest;
import software.amazon.awssdk.services.mediastore.model.CreateContainerResponse;
import software.amazon.awssdk.services.mediastore.model.MediaStoreException;
// snippet-end:[mediastore.java2.create_container.import]

/**
 * Before running this Java V2 code example, set up your development
 * environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class CreateContainer {
    public static long sleepTime = 10;

    public static void main(String[] args) {
        final String usage = """

                Usage:    <containerName>

                Where:
                   containerName - The name of the container to create.
                """;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String containerName = args[0];
        Region region = Region.US_EAST_1;
        MediaStoreClient mediaStoreClient = MediaStoreClient.builder()
                .region(region)
                .build();

        createMediaContainer(mediaStoreClient, containerName);
        mediaStoreClient.close();
    }

    public static void createMediaContainer(MediaStoreClient mediaStoreClient, String containerName) {
        try {
            CreateContainerRequest containerRequest = CreateContainerRequest.builder()
                    .containerName(containerName)
                    .build();

            CreateContainerResponse containerResponse = mediaStoreClient.createContainer(containerRequest);
            String status = containerResponse.container().status().toString();
            while (!status.equalsIgnoreCase("Active")) {
                status = DescribeContainer.checkContainer(mediaStoreClient, containerName);
                System.out.println("Status - " + status);
                Thread.sleep(sleepTime * 1000);
            }

            System.out.println("The container ARN value is " + containerResponse.container().arn());
            System.out.println("Finished ");

        } catch (MediaStoreException | InterruptedException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[mediastore.java2.create_container.main]

File: kotlin/services/mediastore/src/main/kotlin/com/kotlin/mediastore/CreateContainer.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.kotlin.mediastore

// snippet-start:[mediastore.kotlin.create_container.import]
import aws.sdk.kotlin.services.mediastore.MediaStoreClient
import aws.sdk.kotlin.services.mediastore.model.CreateContainerRequest
import kotlinx.coroutines.delay
import kotlin.system.exitProcess
// snippet-end:[mediastore.kotlin.create_container.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args: Array<String>) {
    val usage = """
    
        Usage: <containerName> 

        Where:
               containerName - the name of the container.
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val containerName = args[0]
    println("Status is " + createMediaContainer(containerName))
}

// snippet-start:[mediastore.kotlin.create_container.main]
suspend fun createMediaContainer(containerNameVal: String?) {
    val sleepTime: Long = 10
    val request =
        CreateContainerRequest {
            containerName = containerNameVal
        }

    MediaStoreClient { region = "us-west-2" }.use { mediaStoreClient ->
        val containerResponse = mediaStoreClient.createContainer(request)
        var status = containerResponse.container?.status.toString()

        // Wait until the container is in an active state.
        while (!status.equals("Active", ignoreCase = true)) {
            status = checkContainer(containerNameVal).toString()
            println("Status - $status")
            delay(sleepTime * 1000)
        }
        println("The container ARN value is ${containerResponse.container?.arn}")
        println("The $containerNameVal is created")
    }
}
// snippet-end:[mediastore.kotlin.create_container.main]
